lib/api/core/validation/index.js

```diff
@@ -0,0 +1,788 @@
+        }
+
+        fieldValues = [documentSubset[fieldName]];
+      }
+
+      if (this.types[field.type].allowChildren) {
+        nestedStrictness = this.types[field.type].getStrictness(field.typeOptions, strictness);
+      }
+
+      for (i = 0; i < fieldValues.length; i++) {
+        fieldErrors = [];
+
+        if (!this.types[field.type].validate(field.typeOptions, fieldValues[i], fieldErrors)) {
+          fieldErrors.forEach(message => addErrorMessage(field.path, errorMessages, message, structuredError));
+          return false;
+        }
+
+        if (this.types[field.type].allowChildren && typeof fieldValues[i] === 'object' && field.children) {
+          try {
+            this.recurseFieldValidation(
+              documentSubset[fieldName],
+              originalBodySubset.hasOwnProperty(fieldValues) ? originalBodySubset[fieldValues] : null,
+              field.children,
+              nestedStrictness,
+              errorMessages
+            );
+          }
+          catch (error) {
+            if (error.message === 'strictness') {
+              addErrorMessage(field.path, errorMessages, 'The field validation is strict; it can not add unspecified sub-fields.', structuredError);
+            } else {
+              addErrorMessage(field.path, errorMessages, error.message, structuredError);
+            }
+
+            return false;
+          }
+        }
+      }
+    }
+    else if (field.defaultValue) {
+      if (originalBodySubset !== null) {
+        originalBodySubset[fieldName] = field.defaultValue;
+      }
+
+      documentSubset[fieldName] = field.defaultValue;
+    }
+
+    return reductionResult;
+  }, true);
+};
+
+/**
+ * @returns {Promise.<T>}
+ */
+Validation.prototype.curateSpecification = function () {
+  var
+    promises = [],
+    specification = {};
+
+  return getValidationConfiguration(this.kuzzle)
+    .then(validation => {
+      this.rawConfiguration = validation;
+
+      _.each(this.rawConfiguration, (indexSpec, indexName) => {
+        _.each(indexSpec, (collectionSpec, collectionName) => {
+          var promise = new Promise((resolve) => {
+            this.curateCollectionSpecification(indexName, collectionName, collectionSpec)
+              .then(curatedSpec => {
+                if (!specification.hasOwnProperty(indexName)) {
+                  specification[indexName] = {};
+                }
+
+                specification[indexName][collectionName] = curatedSpec;
+
+                this.kuzzle.pluginsManager.trigger('log:info', `Validation specification for ${indexName} / ${collectionName} has been loaded.`);
+                return resolve({});
+              })
+              .catch(error => {
+                this.kuzzle.pluginsManager.trigger('log:error', `Specification for the collection ${collectionName} triggered an error`);
+                this.kuzzle.pluginsManager.trigger('log:error', `Error: ${error.message}`);
+                return resolve({});
```

Comment:
We resolve to avoid Promise.all to reject (line 335), and doesn't prevent Kuzzle to start.